pkg/util/util.go

```diff
@@ -807,6 +807,16 @@ func IsAzurePlatform() bool {
 	return isAzure
 }
 
+// IsGCPPlatform returns true if this cluster is running on GCP
+func IsGCPPlatform() bool {
+	nodesList := &corev1.NodeList{}
+	if ok := KubeList(nodesList); !ok || len(nodesList.Items) == 0 {
```

Comment:
It's not your PR that did that, but we ask for node list multiple times in every reconcile which seems like a noisy neighbor, so this seems like a good opportunity to refactor and move the node listing and platform detection once in the beginning of reconcile, or even just once in the process lifetime. WDYT?

Reply:
I mean Yeah, That sounds reasonable!